### PR TITLE
guarddog: bazel build files and duration atomic type for portability.

### PR DESCRIPTION
**Pull request by htuch**
While building with various versions of Clang, duration worked better than time_point.

### CODE REVIEW

**Comment by htuch** [comment]
@dnoe @mattklein123 Needed for Bazel+Clang goodness.

File: source/server/watchdog_impl.h [hunk 1]
```diff
@@ -24,21 +18,24 @@ class WatchDogImpl : public WatchDog {
...
-  void touch() override { latest_touch_time_.store(time_source_.currentSystemTime()); }
+  void touch() override {
+    latest_touch_time_.store(time_source_.currentSystemTime().time_since_epoch());
+  }
 
 private:
   const int32_t thread_id_;
   SystemTimeSource& time_source_;
-  std::atomic<SystemTime> latest_touch_time_;
+  std::atomic<std::chrono::system_clock::duration> latest_touch_time_;
```
Comment (mattklein123) [inline source/server/watchdog_impl.h]: nit: Can we rename this latest_touch_time_since_epoch_ (or similar)?

File: source/server/watchdog_impl.h [hunk 2]
```diff
@@ -24,21 +18,26 @@ class WatchDogImpl : public WatchDog {
...
   const int32_t thread_id_;
   SystemTimeSource& time_source_;
-  std::atomic<SystemTime> latest_touch_time_;
+  std::atomic<std::chrono::system_clock::duration> latest_touch_time_since_epoch_;
   Event::TimerPtr timer_;
   const std::chrono::milliseconds timer_interval_;
 };
 
-} // Server
+} // namespace Server
```
Comment (mattklein123) [inline source/server/watchdog_impl.h]: We keep going back and forth on this. Are we doing the "namespace" for the end namespace comments? I don't care either way but we should maybe try to clean this up elsewhere also.

Comment (htuch) [inline source/server/watchdog_impl.h]: This is actually my clang-format doing it automatically. According to https://clang.llvm.org/docs/ClangFormatStyleOptions.html we can flip this on with FixNamespaceComments.

Comment (mattklein123) [inline source/server/watchdog_impl.h]: Not sure if that is supported via clang 3.6 that we are using in CI right now. If clang-format can do this for us that would be fantastic. We might want to think about upgrading clang-format to get more features.

Comment (htuch) [inline source/server/watchdog_impl.h]: Tracking in #673

File: test/server/guarddog_impl_test.cc [hunk 3]
```diff
@@ -208,17 +208,15 @@ TEST(WatchDogBasicTest, ThreadIdTest) {
   gd.stopWatching(watched_dog);
 }
 
-// If this test fails it is because the SystemTime object has become nontrivial
-// or we are compiling under a compiler and library combo that makes the
-// SystemTime object require a lock to be atomicly modified.
+// If this test fails it is because the std::time_t type has become nontrivial or we are compiling
```
Comment (dnoe) [inline test/server/guarddog_impl_test.cc]: Comment nit: std::time_t is generally the legacy "seconds since epoch" but is actually ill-defined in the C++ spec.  Implied is that it is only second granularity like the old C type so not suitable for this application.

File: test/server/guarddog_impl_test.cc [hunk 4]
```diff
@@ -208,17 +208,15 @@ TEST(WatchDogBasicTest, ThreadIdTest) {
...
-// The WatchDog/GuardDog relies on this being a lock free atomic for perf
-// reasons so some workaround will be required if this test starts failing.
+// The WatchDog/GuardDog relies on this being a lock free atomic for perf reasons so some workaround
+// will be required if this test starts failing.
 TEST(WatchDogTimeTest, AtomicIsAtomicTest) {
-  // To compile this test the atomic SystemTime must be non-default initialized,
-  // even though we only call the is_lock_free() method.
   ProdSystemTimeSource time_source;
-  std::atomic<SystemTime> atomic_time(time_source.currentSystemTime());
+  std::atomic<std::chrono::system_clock::duration> atomic_time;
```
Comment (dnoe) [inline test/server/guarddog_impl_test.cc]: Was this change required because under Clang time_point is not lock free atomic but duration is?

File: test/server/guarddog_impl_test.cc (hunk 3 above)
Comment (htuch) [inline test/server/guarddog_impl_test.cc]: Yeah, I fixed that before I shipped, it reflected some earlier understanding of `std::chrono::system_clock`.

File: test/server/guarddog_impl_test.cc (hunk 4 above)
Comment (htuch) [inline test/server/guarddog_impl_test.cc]: It's not the lock free issue, it was a compile issue related to the `std::atomic` type in `latest_touch_time_since_epoch_ `. It happened when I was using Clang 3.4 with an earlier libstdc++, we're now working with a more recent version of Clang so it might not be a problem. It seems duration is a safe bet, it's easy to convert to milliseconds `uint64_t` if it ever causes a problem in the future as well.
